Sources/Moya/RawDataParameterEncoder.swift

```diff
@@ -0,0 +1,21 @@
+import Foundation
+import Moya
+
+/// An "encoder" that expects the given parameters to be of type `Data` and just sets the request's httpBody with it, without any additional encoding.
+struct RawDataParameterEncoder: ParameterEncoder {
+    func encode<Parameters>(_ parameters: Parameters?, into request: URLRequest) throws -> URLRequest where Parameters: Encodable {
+        guard let parameters = parameters else { return request }
+
+        var request = request
```

Comment:
This is more of a question,  if we are assign the request to a mutating, modifying and then returning... could be a valid approach reveice this request here as `inout` and modify it directly?

Reply:
It would indeed be nice. But if we add the inout keyword in the function definition it is no longer the one from `ParameterEncoder`, raising the error `Type 'RawDataParameterEncoder' does not conform to protocol 'ParameterEncoder'`